src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -69,6 +70,11 @@ public class QueryInsightsSettings {
      */
     public static final String PLUGINS_BASE_URI = "/_insights";
 
+    public static final GroupingType DEFAULT_GROUPING_TYPE = GroupingType.NONE;
+    public static final int DEFAULT_GROUPS_EXCLUDING_TOPN_LIMIT = 100;
+
+    public static final int MAX_GROUPS_EXCLUDING_TOPN_LIMIT = 10000;
```

Comment:
How much memory would 10000 records consume based on the benchmark results?

Reply:
Did not capture this and we did not reach anywhere close to the 10000 limit in the benchmarks. 

> If a search query record is around 1kb then 10000 groups means we will consume 10mb memory at most for this feature, which should be fine. but we still need to watch out on the memory consumption here for queries with large source.
This analysis seems reasonable but we would need to keep watch out for the memory consumption here.